src/core/elevation/qgsabstractprofilegenerator.h

```diff
@@ -164,6 +165,16 @@ class CORE_EXPORT QgsAbstractProfileResults
      * Snaps a \a point to the generated elevation profile.
      */
     virtual QgsProfileSnapResult snapPoint( const QgsProfilePoint &point, const QgsProfileSnapContext &context );
+
+    /**
+     * Updates the results to match the properties from the specified \a generator.
+     *
+     * For instance, this method can be used to replace any stored properties relating to rendering
+     * the gathered results to reflect the \a generator's current properties.
+     *
+     * The base class method does nothing.
+     */
+    virtual void updateFromGenerator( const QgsAbstractProfileGenerator *generator );
```

Comment:
I am quite unsure about semantics of this call... Right now all implementations just update the appearance-related pieces of data. But the method doxygen description (first sentence) made me think that in an implementation also results data should be overwritten from the given generator.

I wonder if the method could be renamed somehow to make it more clear we're only after copying generator properties...

Reply:
I've renamed to "copyPropertiesFromGenerator" and added clarification to the documentation